src/main/java/fr/sncf/osrd/train/Train.java

```diff
@@ -155,13 +157,27 @@ public void interact(Simulation sim, Signal signal, InteractionType interactionT
                 lastState.setAspectConstraints(signalState);
                 break;
             case HEAD:
-                signalState.unsubscribeTrain();
+                onTrainReachSignal(signalState);
                 break;
             default:
                 throw new RuntimeException("Unexpected signal interaction type");
         }
     }
 
+    private void onTrainReachSignal(SignalState signalState) throws SimulationError {
+        var blocking = signalState.aspects.stream().anyMatch(
+                aspect -> aspect.constraints.stream()
+                        .filter(aspectConstraint -> aspectConstraint instanceof AspectConstraint.SpeedLimit)
+                        .map(aspectConstraint -> (AspectConstraint.SpeedLimit) aspectConstraint)
+                        .anyMatch(speedLimit -> speedLimit.speed <= 0)
```

Comment:
I don't know if we can retrieve the information here, but ideally we could check that the current speed is less or equal to the `aspectConstraints`.